### PR TITLE
fix: logo field in configuration cannot use the local file

### PR DESCRIPTION
**Pull request by jinliming2**
**Type:** bug

**Description:** As [document](https://verdaccio.org/docs/en/webui) says that the logo field in configuration can be a URI with http(s) scheme, but it is more useful the logo field can specify a local file. Besides, also the example in the document seems like using a local file path.

### CODE REVIEW

**Comment by juanpicado** [comment]
@jinliming2 thanks for the PR. But I need something else. I need the why is failing and the steps to reproduce + configurations that make it works with your PR. 

Also any PR we need a unit test, I can guide you in that area if you need my help.

**Comment by jinliming2** [comment]
image

If you specified a local file in configuration, Logo on web UI will be missing.

image

The URL for the image is an absolute path, this path was used for lookup packages.

image

**Comment by jinliming2** [comment]
The configuration works for my PR:
```yaml
web:
  enable: true
  logo: logo.png
```
```yaml
web:
  enable: true
  logo: ./logo.png # Related to the current working directory
```
```yaml
web:
  enable: true
  logo: /absolute/path/to/logo.png
```

**Comment by juanpicado** [comment]
Thank you @jinliming2 that was pretty complete.  Let me run some test in the next few days.

File: src/api/web/index.js [hunk 1]
```diff
@@ -43,6 +43,25 @@ module.exports = function(config, auth, storage) {
   const indexTemplate = path.join(themePath, 'index.html');
   const template = fs.readFileSync(indexTemplate).toString();
 
+  // Logo
+  let logoURI = _.get(config, 'web.logo') ? config.web.logo : '';
+  if (logoURI && !/^(https?:)?\/\//.test(logoURI)) {
+    // URI that not starts with "http://", "https://" or "//"
+    logoURI = path.join('/-/static/', path.basename(logoURI));
+    router.get(logoURI, function(req, res, next) {
+      res.sendFile(path.resolve(config.web.logo), function(err) {
```
Comment (juanpicado) [inline src/api/web/index.js]: I like your solution, but we need to clean up this a bit. The following callback is duplicated in the same file. 
```
function(err) {
        if (!err) {
          return;
        }
        if (err.status === HTTP_STATUS.NOT_FOUND) {
          next();
        } else {
          next(err);
        }
      }
```

File: src/api/web/index.js [hunk 2]
```diff
@@ -43,6 +43,25 @@ module.exports = function(config, auth, storage) {
   const indexTemplate = path.join(themePath, 'index.html');
   const template = fs.readFileSync(indexTemplate).toString();
 
+  // Logo
+  let logoURI = _.get(config, 'web.logo') ? config.web.logo : '';
+  if (logoURI && !/^(https?:)?\/\//.test(logoURI)) {
```
Comment (juanpicado) [inline src/api/web/index.js]: This `!/^(https?:)?\/\//.test(logoURI)` should be on utilities `src/lib/utils.js`. Export a function so we can reuse it later.

Comment (juanpicado) [reply src/api/web/index.js]: Please include couple of unit test. `test/unit/api/utils.spec.js`

**Comment by juanpicado** [comment]
One more thing
- [ ] You would need update documentation with all different set of options purposed https://github.com/verdaccio/website/blob/master/docs/web.md

File: src/api/web/index.js (hunk 1 above)
Comment (jinliming2) [reply src/api/web/index.js]: Done.

File: src/api/web/index.js (hunk 2 above)
Comment (jinliming2) [reply src/api/web/index.js]: Done.

**Comment by juanpicado** [comment]
@jinliming2 one thing, we need update here https://github.com/verdaccio/website/blob/master/docs/web.md so all are aware of all set of options to be used

**Comment by juanpicado** [comment]
@jinliming2 👏 released https://github.com/verdaccio/verdaccio/releases/tag/v4.0.0-beta.3

**Comment by jinliming2** [comment]
Perfect! 👍

**Pull request merged**